router/src/validation.rs

```diff
@@ -266,10 +271,19 @@ impl Validation {
             .unwrap_or(Ok(None))?;
 
         // Validate inputs
-        let (inputs, input_length, max_new_tokens) = self
+        let (inputs, _input_length, max_new_tokens) = self
             .validate_input(request.inputs, truncate, max_new_tokens)
             .await?;
 
+        let (negative_inputs, input_length, max_new_tokens) = self
+            .validate_input(negative_inputs, truncate, Some(max_new_tokens))
+            .await?;
```

Comment:
can we avoid calling `validate_input` if negative inputs is empty? 

```suggestion
        let (negative_inputs, input_length, max_new_tokens) = if !negative_inputs.is_empty() {
            self.validate_input(negative_inputs, truncate, Some(max_new_tokens))
                .await?
        } else {
            (String::new(), _input_length, max_new_tokens)
        };
```